### PR TITLE
Add BLS support to grub_menuentry

### LINKED ISSUE #37: The grub2 system should update both the EFI and non-EFI configurations when triggered
This was an oversight in my original implementation but systems that can booth either EFI or legacy will potentially switch between the two at will.

This means that both configuration paths need to be updated.

At this point, I'm thinking that it may be best to update *all* non-symlink copies of target grub configuration files on the system so that random changes don't surprise us in the future but would like some feedback on this idea.

### LINKED ISSUE #38: Absent GRUB_CMDLINE_LINUX_DEFAULT can result in duplicated kernel parameters.
When using grub2 the initial state in /etc/default/grub includes:

  GRUB_CMDLINE_LINUX populated
  GRUB_CMDLINE_LINUX_DEFAULT absent

Used:

    kernel_parameter { "pci=nomsi,noaer":
        ensure   => present,
        bootmode => "normal",
    }

results in GRUB_CMDLINE_LINUX_DEFAULT being copied from GRUB_CMDLINE_LINUX and then massaged by the lens.

The resultant kernel boot line includes duplicate entries. According to the grub2 manual (https://www.gnu.org/software/grub/manual/grub/grub.html):

   ‘GRUB_CMDLINE_LINUX_DEFAULT’

        Unless ‘GRUB_DISABLE_RECOVERY’ is set to ‘true’, two menu entries will be
        generated for each Linux kernel: one default entry and one entry for recovery
        mode. This option lists command-line arguments to add only to the default
        menu entry, after those listed in ‘GRUB_CMDLINE_LINUX’.

Desired behaviour is for GRUB_CMDLINE_LINUX_DEFAULT to not being populated from GRUB_CMDLINE_LINUX as they maybe concatenated when generating the grub.conf

### LINKED ISSUE #49: grub_menuentry is broken in EL8 and Fedora 30+
EL8 and Fedora 30+ now use the `blscfg` module to load OS configurations from `/boot/loader/entries`. This means that we're going to have to fall back to `grubby` for systems that include this module in their `/etc/grub2.cfg` configuration until some method can be found to process this information without requiring `grubby`.

### PR DESCRIPTION
**Pull request by trevor-vaughan**
* Adds BLS support which should fix issues on EL8 and Fedora 30
* Fixed the issue with duplicating kernel options while debugging
* Updates both non-EFI and EFI paths

Closes #49
Closes #38
Closes #37

### CODE REVIEW

File: lib/puppet/type/grub_menuentry.rb [hunk 1]
```diff
@@ -8,6 +8,9 @@
 # Based on work by Dominic Cleal
 
 Puppet::Type.newtype(:grub_menuentry) do
+  require 'puppet/parameter/boolean'
+  require 'puppet/property/boolean'
```
Comment (alexjfisher) [inline lib/puppet/type/grub_menuentry.rb]: IIRC Puppet::Property::Boolean doesn't work. If you try to set a *property* to the boolean `false`, puppet won't treat it any differently from if you had set it to `undef` or hadn't set it at all.

Comment (alexjfisher) [reply lib/puppet/type/grub_menuentry.rb]: See https://github.com/theforeman/puppet-pulp/commit/7fdb52396ae6200d2dcfefc248e327c5548c25c0

Comment (trevor-vaughan) [reply lib/puppet/type/grub_menuentry.rb]: Well neat :-|. I'll work on getting that updated.

**Comment by trevor-vaughan** [comment]
@alexjfisher This should be good to go now.

**Pull request merged**